### PR TITLE
#579 Bottom bar improvements

### LINKED ISSUE #579: Snackbar appears over bottom toolbar
### Are you ready to report?
   - Your issue may already be reported! Please search on the [issue track](../) before creating one.
   - Ensure you have tested with the last (dev) version of the software.  (If not, and if possible, run your test again with the most recent version available.)

### Environment
  * version of the software you use : Android Studio 3.0.1
  * device / operating system : Nexus 5X / Android 8.1.0
  
### The Bug:

  * What do you obtain? A copy of the error message or a screenshot is always useful.

The snackbar appear over bottom navigation bar when we bookmark / remove bookmark any item.

  * When does this occur? 

#### Steps to reproduce:

  1. Go to setting and enable bottom toolbar.
  2. Now an the main page try bookmarking any article
  3. Try removing the bookmark also.
  ...
  
  * What should be the behaviour from your point of view? How do you expect the service to work?
As per the [Material UI guidelines](https://material.io/guidelines/components/bottom-navigation.html#bottom-navigation-specs) the snackbar should appear above the bottom navigation bar as shown in the screenshot below.

### Other Comments:
  Would like to work on it.

### PR DESCRIPTION
**Pull request by mhutti1**
Fixes #579 #1375

### CODE REVIEW

**Comment by macgills** [comment]
A [Behaviour](https://stackoverflow.com/questions/33217241/how-to-move-a-view-above-snackbar-just-like-floatingbutton) is how we should make views move when a snackbar appears

**Comment by mhutti1** [comment]
The FAB wasn't really the issue that was being solved. It is a different issue with a much lower priority.

**Comment by macgills** [comment]
This has nothing to do with a FAB, I believe the issue was that the snackbar was obscuring the bottom bar. The SO link gives an example of how to animate an arbitrary view up when a snackbar is shown

**Comment by mhutti1** [comment]
Sure but thats not what we want to do. We want to put the snackbar over the bottom bar not the bottom bar over the snackbar. That would look really weird and be bad UX. If a user tried to bookmark then unbookmark they would end up clicking the snackbar not the unbookmark button because the button was moved up.

**Comment by macgills** [comment]
My apologies, didn't read the ticket fully, though this bar is not a BottomNavigationBar the material guidelines should probably be followed

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -124,6 +124,16 @@
...
         />
 
+
+      <androidx.coordinatorlayout.widget.CoordinatorLayout
+        android:id="@+id/snackbar_root"
+        android:layout_width="match_parent"
+        android:layout_height="100dp"
+        app:layout_constraintBottom_toTopOf="@id/activity_main_button_pause_tts"
+        app:layout_constraintEnd_toEndOf="parent"
+        >
```
Comment (macgills) [inline app/src/main/res/layout/activity_main.xml]: make this `/>` so we don't have the other closing tag

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -124,6 +124,16 @@
         tools:visibility="visible"
         />
 
+
+      <androidx.coordinatorlayout.widget.CoordinatorLayout
+        android:id="@+id/snackbar_root"
+        android:layout_width="match_parent"
+        android:layout_height="100dp"
```
Comment (macgills) [inline app/src/main/res/layout/activity_main.xml]: can it be `wrap_content` or do we need to define the height?

**Pull request merged**